projects/rocr-runtime/runtime/hsa-runtime/core/runtime/intercept_queue.cpp

```diff
@@ -126,6 +126,9 @@ InterceptQueue::InterceptQueue(std::unique_ptr<Queue> queue)
          "Packet intercept error: initial retry index is incompatible with IsPendingRetryPoint.\n");
   buffer_ = SharedArray<AqlPacket, 4096>(wrapped->amd_queue_.hsa_queue.size);
   amd_queue_.hsa_queue.base_address = reinterpret_cast<void*>(&buffer_[0]);
+  
+  // Pre-allocate staging buffer with queue size
+  staging_buffer_.resize(wrapped->amd_queue_.hsa_queue.size);
```

Comment:
this is too big, you dont need to resize to entire queue size. I would say max 256 elements is good enough